### PR TITLE
net package: linux traffic stats

### PR DESCRIPTION
**Pull request by servetozkan**

### CODE REVIEW

**Comment by shirou** [comment]
gopsutil is a multi-platform library. so please add at least windows, darwin, and FreeBSD implementation if you want to add a feature. (return ` return 0, common.ErrNotImplementedError` is ok.)

**Comment by servetozkan** [comment]
Check now

File: net/net_linux.go [hunk 1]
```diff
@@ -16,8 +16,92 @@ import (
...
+func TrafficStats(pernic bool, interval time.Duration) ([]TrafficStat, error) {
+	stats1, err := TrafficStatsWithContext(context.Background(), pernic)
+	time.Sleep(interval * time.Second)
+	stats2, err := TrafficStatsWithContext(context.Background(), pernic)
+
+	merged, err := CalculateTraffic(stats1, stats2, interval)
+	return merged, err
+}
+
+func CalculateTraffic(stats1 []TrafficStat, stats2 []TrafficStat, interval time.Duration) ([]TrafficStat, error) {
```
[the remark this answers is not in the document]
Comment (Lomanic) [reply net/net_linux.go]: Please unexport this function.

**Comment by servetozkan** [comment]
The requested changes have been completed.

**Comment by Lomanic** [comment]
As-is this PR shouldn't be merged and may even be closed.

As I'm not too familiar with this part of the code I didn't see at first that it was based off the IOCounters* functions (that should have been mentioned, and some explication text should really have been added to tell the intent of the PR ; even better a feature request would have been preferred to discuss the implementation before sending code), the TrafficStats* functions are just trimmed down version of their IOCounters* counterpart. If we were to add this new TrafficStat, it would simply be calculated using the IOCounters* functions instead of reimplementing them by copy-paste.

I understand why it would be useful for end-users to be able to obtain traffic speed of their network cards during a given interval without implementing this logic themselves, but is it sensible to add a new type for that? This would create some duplication with IOCounters that already includes BytesSent and BytesRecv information.

So regarding the exposed API:

1. shall we add in IOCountersStat type new *Bytes*RecvPerSecond and *Bytes*SentPerSecond fields (so we have bytes-based types everywhere), and create new IOCounters* functions that would take a `duration` argument and that would properly populate these two fields? This won't break backward compatibility as we're adding new fields/functions.
2. shall we add this new TrafficStat type (while internally calling IOCounters* functions instead of reimplementing them like currently) with new functions (all the TrafficStats* functions) as in current PR? If user is interested in both IOCountersStat and TrafficStat information their program would call twice the IOCounters* functions to get all information which is suboptimal.

Either way, this should be defined in `net/net.go` as it is platform-agnostic (relying on OS-specific IOCounters* functions).

I prefer the first option (better optimized and looks more coherent to me).

**Comment by shirou** [comment]
I almost agree with @Lomanic . I think a library doesn't expose such a complex, state required API. A library should keep "simple", not "easy". Library use can build their logic very quick by using current gopsutil. (`load avg` is a special case because it is widely used from ancient)

**Comment by Lomanic** [comment]
@shirou  if you think this should still be implemented in the lib, don't hesitate to create an issue to discuss the implementation.

**Pull request closed without merging**